libkbfs/tlf_handle_resolve.go

```diff
@@ -769,12 +776,33 @@ func parseTlfHandleLoose(
 		}
 	}
 
+	canonicalName := string(h.GetCanonicalName())
 	if extensionSuffix != "" {
 		extensionList := tlf.HandleExtensionList(extensions)
 		sort.Sort(extensionList)
-		var canonExtensionString = extensionList.Suffix()
+		var canonExtensionString string
+		// If this resolve is being done in a "quick" way that doesn't
+		// check for implicit teams, we might not know if this handle
+		// is backed by a team or not.  But (mostly for tests) we need
+		// to make sure the suffix reflects the right format,
+		// otherwise we'll get too many levels of non-canonical
+		// redirects.  So if the user explicitly specified a "#1" in
+		// their suffix, let's keep it there and assume the user knows
+		// what they're doing.
+		if h.IsBackedByTeam() || strings.Contains(extensionSuffix, "#1") {
```

Comment:
Why is this whitelisting `#1`? What about `#2`? `TestSBSConflicts` generates more than one conflict for each TLF.

Reply:
The whole issue is that, historically, we have not shown `#1` for non-team conflict TLFs, only showing the number starting at `#2`.  But max wanted it to be simpler for teams, so we always show the number now, even `#1`.  Ideally we would just switch non-team TLFs to do the same thing, but we can't because that old string is saved in places like chats and needs to remain the same going forward.  So this whole PR is just about dealing with the `#1` discrepancy, and making it show up only when it is supposed to.

Reply:
OK I think I follow now. What is supposed to happen if the user passed a `#1` but it's not an implicit team?

Reply:
Ideally, it would give back a non-canonical name error leading it to the one without the "#1".  But this PR breaks that behavior.  I don't really think it's worth worrying about though, as I'm not sure why anyone would ever make a folder name like that -- they'd have to type it in manually.